### PR TITLE
Bring back section chunking

### LINKED ISSUE #276: Bugg: RangeError: Invalid string length
🐛 When adding long reports we get a RangeError from ChromaDb
⚡ A potential fix is to introduce batching to chroma since it seem to be happening for longer reports

Ex: https://group.vattenfall.com/globalassets/corporate/who-we-are/sustainability/vattenfall-annual-and-sustainability-report-2023.pdf

### PR DESCRIPTION
**Pull request by hugo-nl**
🎸 Fix chroma range error bugg
🎸 improved reduce funciton to vision api

Fix #276

### CODE REVIEW

**Review by irony — APPROVED** [review]
co-authored by me so approved :)

File: src/workers/indexMarkdown.ts [hunk 1]
```diff
@@ -10,20 +10,16 @@ const indexMarkdown = new DiscordWorker(
...
-    const overlapSize = 200
-
-    const chunks: string[] = []
-    for (let i = 0; i < markdown.length; i += chunkSize - overlapSize) {
-      const chunk = markdown.slice(i, i + chunkSize).trim()
-      if (chunk.length > 0) chunks.push(chunk)
-    }
+    const chunks = markdown
+      .split('\n###')
+      .map((p) => p.trim())
```
Comment (irony) [inline src/workers/indexMarkdown.ts:15]:
```suggestion
      .map((p) => '\n###' + p.trim())
```
vi slänger bort en radbrytning så det behöver vi lägga tillbaka

**Review by irony — APPROVED** [review]

File: src/workers/nlmExtractTables.ts [hunk 2]
```diff
@@ -104,27 +104,26 @@ const nlmExtractTables = new DiscordWorker(
...
-          )
+      const tables = await Object.entries(groupedResults).reduce(
+        async (resultsPromise, [page_idx, tablesOnPage]) => {
+          const results = await resultsPromise
+          const lastPageMarkdown = results.at(-1)?.markdown || ''
           const filename = tablesOnPage[0].filename
           const markdown = await extractTextViaVisionAPI(
             { filename, name: `Tables from page ${page_idx}` },
-            context.slice(-3).join('\n')
+            lastPageMarkdown.slice(0, 5000)
```
Comment (Greenheart) [inline src/workers/nlmExtractTables.ts:114]: Why limit to 5000 characters here?

Does this mean that we send a larger context compared to before when we only sent the last 3 tables?

Comment (irony) [inline src/workers/nlmExtractTables.ts:114]:
```suggestion
            lastPageMarkdown
```

**Review by irony — APPROVED** [review]

**Pull request merged**